lib/index.js

```diff
@@ -73,37 +75,283 @@ TwitterAds.prototype.page = function() {
  */
 
 TwitterAds.prototype.track = function(track) {
-  var events = this.events(track.event());
-  var self = this;
-  each(events, function(pixelId) {
-    self.load('singleTag', defaults({
-      pixelId: pixelId,
-      quantity: track.proxy('properties.quantity'),
-      revenue: track.revenue()
-    }, defaultQueryValues));
-  });
+  this.fireLegacyConversionTags(track);
+};
+
+/**
+ * Products Searched
+ *
+ * Required params: None
+ *
+ * @api public
+ * @param {Track} track
+ */
+
+TwitterAds.prototype.productsSearched = function(track) {
+  if (this.options.universalTagPixelId) {
+    var payload = setStatus(track.properties());
+    window.twq('track', 'Search', payload);
+  }
+};
+
+/**
+ * Product Viewed
+ *
+ * Required params: `content_ids`, `content_type`
+ *
+ * @api public
+ * @param {Track} product
+ */
+
+TwitterAds.prototype.productViewed = function(product) {
+  if (this.options.universalTagPixelId) {
+    var props = product.properties();
+    var identifier = this.options.identifier; // sku or productId
+    var payload = {
+      content_ids: [product[identifier]()],
+      content_type: 'product', // Must be hardcoded
+      content_name: product.name(),
+      content_category: product.category(),
+      currency: product.currency() // Fallsback to 'USD'
+    };
+
+    payload = extend(payload, formatValue(props));
+    payload = extend(payload, setStatus(props));
+
+    window.twq('track', 'ViewContent', payload);
+  }
+};
+
+/**
+ * Product Added
+ *
+ * Required params: `content_ids`, `content_type`
+ *
+ * @api public
+ * @param {Track} product
+ */
+
+TwitterAds.prototype.productAdded = function(product) {
```

Comment:
ohhhhhh! that's right. yep I can just add the `fireLegacyConversionEvents` inside each of these functions in case they had mapped any of these events as single event tags @sperand-io